src/main/java/com/iota/iri/service/API.java

```diff
@@ -400,7 +401,7 @@ private AbstractResponse checkConsistencyStatement(List<String> transactionsList
         if (state) {
             snapshotProvider.getLatestSnapshot().lockRead();
             try {
-                WalkValidatorImpl walkValidator = new WalkValidatorImpl(tangle, snapshotProvider, ledgerService, configuration);
+                WalkValidator walkValidator = new WalkerValidatorNoMaxDepthImpl(tangle, snapshotProvider, ledgerService, configuration);
```

Comment:
Could you not simply contain a separate Overloaded function in the `WalkValidatorImpl` to avoid the creation of a whole class just for `checkConsistency` calls? @galrogo